docs/design/map/map-store-operation-offloading.md

```diff
@@ -0,0 +1,229 @@
+          -------------------------------------------------------
+                        Create steps from operation 
+          --------------------------------------------------------      
+                                  |
+          -------------------------------------------------------
+                    Run step on partition thread.                 <-----+
+          --------------------------------------------------------      |
+                                  |                                     | 
+                                  |                                     |
+                                  |                                     |
+          ----------------------------------------------------    If No |
+                  Do we encounter offload needed Step?      ------------+
+          ----------------------------------------------------
+                                  | 
+                                  | If Yes
+                                  |
+          -------------------------------------------------------
+                  Offload next Step and free partition thread
+          -------------------------------------------------------
+                                  | 
+                                  |
+                          --------------------
+                            End of execution
+                          --------------------
+
+
+#### Flow in Offloaded Thread
+          -------------------------------------------------------
+            Run offload needed Step of the operation here         <-----+
+          --------------------------------------------------------      |
+                                  |                                     | 
+                                  |                                     |
+          ----------------------------------------------------   If Yes |
+                  Do we encounter offload needed Step?      ------------+
+          ----------------------------------------------------
+                                  | 
+                                  | If No
+                                  |
+          -------------------------------------------------------
+              Schedule next Step to run in partition thread
+          -------------------------------------------------------
+                                  | 
+                                  |
+                          --------------------
+                            End of execution
+                          --------------------
+
+### Risks and Assumptions
+All MapStore interacting operations are queued per map per
+partition and next operations must wait end of head-operation
+before execution, although this is an improvement when we
+compare it with the current situation(which is all operations
+of all maps in a partition waiting head-operation), it can
+also be deemed as a limitation to preserve current behavior.
```

Comment:
Could you also note the relationship of this offloaded operations with the operations that do not interact with the map store? e.g. `map.getEntryView(K key)`

Reply:
done.